sys/include/net/sock/async/event.h

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (C) 2019 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup        net_sock_async_event    Asynchronous sock with event API
+ * @ingroup         net_sock
+ * @brief           Provides an implementation of asynchronous sock for
+ *                  @ref sys_event
+ * @experimental    This API is still under development and should not be used
+ *                  in production yet.
+ *
+ * How To Use
+ * ----------
+ *
+ * You need to [include](@ref including-modules) at least one module that
+ * implements a [`sock` API](@ref net_sock) and the module `sock_async_event` in
+ * your application's Makefile.
```

Comment:
So

- `gnrc_sock_ip`
- `gnrc_sock_udp`
- `lwip_sock_ip`
- `lwip_sock_tcp`
- `lwip_sock_udp`
- `tinydlis_sock_dtls`
- ...

This could become quite an extensive list and would have to be amended everytime a new port for the sock API is provided (so also a potential merge conflict hassle like any list that is changing) for *every* sock doc that contains this sentence...

Reply:
Sorry I meant giving the list of `USEMODULE` for the `sock_udp` example only.

Reply:
That still causes the same problem, just with a smaller list.

Reply:
I used the following modules:

```
USEMODULE += auto_init_gnrc_netif
USEMODULE += gnrc_netdev_default
USEMODULE += gnrc_ipv6_default
USEMODULE += sock_async_event
USEMODULE += gnrc_sock_async
USEMODULE += gnrc_sock_udp
```

I think it would make easier for a user to give it a test if those modules are given es examples. What I mean by that would be for example `and at least one network device (e.g: gnrc_netdev_default)` and `implements a [`sock` API](@ref net_sock) (e.g: gnrc_sock_async, gnrc_sock_udp)`. I won't insist on this though, since it wasn't hard to figure out the needed `MODULES`

```

Reply:
> I think it would make easier for a user to give it a test if those modules are given es examples. What I mean by that would be for example `and at least one network device (e.g: gnrc_netdev_default)` and `implements a [`sock` API](@ref net_sock) (e.g: gnrc_sock_async, gnrc_sock_udp)`. I won't insist on this though, since it wasn't hard to figure out the needed `MODULES`

I agree with the examples for `sock`, but I don't think mentioning that a network device is necessary for network communication is valuable information here (you also didn't mention that a network layer protocol implementation is required... you see how complicated things can become if you explain _**everything** everywhere_? ;-))

Reply:
Hahaha ok, lets not go down that road.